bin/MadGraph5_aMCatNLO/cards/production/13TeV/HToZATo2L2B_ggfusion_b-associatedproduction/example_cards/HToZATo2L2B_500p00_300p00_1p50_ggH_TuneCP5_13TeV_pythia8/HToZATo2L2B_500p00_300p00_1p50_ggH_TuneCP5_13TeV_pythia8_run_card.dat

```diff
@@ -40,43 +40,46 @@
 # PDF CHOICE: this automatically fixes also alpha_s and its evol.    *
 #*********************************************************************
   lhapdf	= pdlabel ! PDF set                                     
-$DEFAULT_PDF_SETS = lhaid ! if pdlabel=lhapdf, this is the lhapdf number
+325500 = lhaid ! if pdlabel=lhapdf, this is the lhapdf number
```

Comment:
Why is this needed. Why does default not work for you?

Reply:
I am using this lhapdf because it's recommended for UL campaign [ 1] as well i am following this paper [2] for simulation the b-associated production and they're using the same pdfsets  metionned here. 
By the way, I did test my settings with the defaults pdf : 
```
  lhapdf	= pdlabel ! PDF set
$DEFAULT_PDF_SETS = lhaid ! if pdlabel=lhapdf, this is the lhapdf number
              ! numbers for central PDF sets are allowed. Can be a list; 
              ! PDF sets beyond the first are included via reweighting.
```
I can see that the same pdf sets have been included in my ``reweight_xsec_events.output`` when using the default pdfsets for my bbH process : 
```
==== LHAPDF6 USING DEFAULT-TYPE LHAGLUE INTERFACE ====
LHAPDF 6.2.1 loading /cvmfs/cms.cern.ch/slc7_amd64_gcc700/external/lhapdf/6.2.1-pafccj3/share/LHAPDF/NNPDF31_nnlo_as_0118_nf_4_mc_hessian/NNPDF31_nnlo_as_0118_nf_4_mc_hessian_0000.dat
NNPDF31_nnlo_as_0118_nf_4_mc_hessian PDF set, member #0, version 1; LHAPDF ID = 325500
Thanks for using LHAPDF 6.2.1. Please make sure to cite the paper:
  Eur.Phys.J. C75 (2015) 3, 132  (http://arxiv.org/abs/1412.7420)
 A PDF is used, so alpha_s(MZ) is going to be modified
 Old value of alpha_s from param_card:   0.11799999999999999     
 New value of alpha_s from PDF lhapdf :  0.11266013434007299     
 Enter event file name
 Enter 1 to save all cross sections on tape
       0 otherwise
   
 The events appear to be unweighted
  Will store the ratios of recomputed weights
  over reference weights
 ================================
 process combination map (specified per FKS dir):
  1 map         1
  1 inv. map    1
  2 map         1
  2 inv. map    1
  3 map         1
  3 inv. map    1
  4 map         1
  4 inv. map    1
  5 map         1   1   1   1
  5 inv. map    4
  6 map         1   1   1   1
  6 inv. map    4
  7 map         1   1   1   1
  7 inv. map    4
  8 map         1   1   1   1
  8 inv. map    4
 ================================
```
Please have a look to the gridpack with the default pdf setting [3] 
To be honest, I didn't study these pdfsets, I am just following the recommendation I found. If you think that's `325500 = lhaid` is not a wise a choice and better keep the default. I will stick with the default then ! 

[1] https://cms-pdmv.gitbook.io/project/mccontact/info-for-mc-production-for-ultra-legacy-campaigns-2016-2017-2018
[2] https://arxiv.org/pdf/1808.01660.pdf
[3] ```/afs/cern.ch/user/k/kjaffel/public/ZArun2_gridpacks/compare_bbHwidths/HToZATo2L2B_800p00_700p00_20p00_bbH4F_TuneCP5_13TeV-amcatnlo_pythia8_slc7_amd64_gcc700_CMSSW_10_6_19_tarball.tar.xz```

Reply:
What I mean: If you leave Default_PDF_Set you should get 325500 if everything is ok. If not you will not and you know something is wrong with your setup. If you hard-code it, you prevent this check. Hence my proposal ...

Reply:
What Alexander is saying, is that if you use $DEFAULT_PDF_SET, it will still give you the same PDF set (325500) + some additional other PDF sets written in [1] (the first line 325500 being the nominal PDF). And because this is CMS specific setup, using this you can check your gridpack setups just in case.

[1] https://github.com/cms-sw/genproductions/blob/master/MetaData/pdflist_4f_2017.dat

Reply:
Yes, I totally agree,  [solved here ](https://github.com/cms-sw/genproductions/pull/2884#issuecomment-850299058).